Enrich_methods/material_db_tools.py

```diff
@@ -0,0 +1,444 @@
+        density (float): The density of the material.
+        citation (str): A citation or reference for the material data.
+        molecular_mass (Optional[float]): The molecular mass of the material, if applicable.
+        mass_enrichment (Optional[Dict[int, Dict[int, float]]]): A dictionary containing the
+            enrichment information for elements, where keys are element IDs (int) and values are
+            dictionaries with nuclide IDs (int) as keys and enrichment fractions (float) as values.
+
+    Returns:
+        Material: The created Material object with the specified properties, based on mass fractions.
+    """
+    if mass_enrichment:
+        for element, enrichment_vector in mass_enrichment.items():
+            enriched_mat = Material(enrichment_vector)
+            nucvec = enrich(nucvec, element, enriched_mat.comp)
+    mat = Material(nucvec, density=density, metadata={"citation": citation})
+    if molecular_mass:
+        mat.molecular_mass = molecular_mass
+    return mat.expand_elements()
+
+
+def make_mat_from_atom(
+    atom_frac: Dict[Union[int, str], float],
+    density: float,
+    citation: str,
+    mass_enrichment: Optional[Dict[int, Dict[int, float]]] = None,
+) -> Material:
+    """
+    Create a Material object from atom fraction data.
+
+    Args:
+        atom_frac (Dict[Union[int, str], float]): The atomic fraction composition, with keys as
+            nuclide IDs (zas) or the pyne specified element name and values as their fractions (float).
+        density (float): The density of the material.
+        citation (str): A citation or reference for the material data.
+        mass_enrichment (Optional[Dict[int, Dict[int, float]]]): A dictionary containing the
+            enrichment information for elements, where keys are element IDs (int) and values are
+            dictionaries with nuclide IDs (int) as keys and enrichment fractions (float) as values.
+
+    Returns:
+        Material: The created Material object with the specified properties, based on atomic fractions.
+    """
+    if mass_enrichment:
+        for element, enrichment_vector in mass_enrichment.items():
+            enriched_mat = Material(enrichment_vector)
+            atom_frac = enrich(atom_frac, element, enriched_mat.to_atom_frac())
+
+    mat = Material()
+    mat.from_atom_frac(atom_frac)
+    mat.density = density
+    mat.metadata["citation"] = citation
+    return mat.expand_elements()
+
+
+def get_consituent_citations(materials):
+    """
+    Retrieve citations from a list of materials.
+
+    Args:
+        materials (list): A list of Material objects.
+
+    Returns:
+        str: A comma-separated string of citations.
+    """
+    citation_list = []
+    for mat in materials:
+        citation_list.append(mat.metadata["citation"])
+
+    return ", ".join(citation_list)
+
+
+# Mix Materials by Volume
+def mix_by_volume(material_library, vol_fracs, citation, mass_enrichment=None):
```

Comment:
I think my problem with this is that it disrupts the cognitive flow of a simple mixing method.

What if you had a method that did the work of making the new enriched materials, and called that from the beginnng of this method, adding those materials to the library, and then the rest of the method would largely be the same as it originally was?

```
# pseudocode and not exactly right....
def mix_by_volume(material_library, vol_fracs, citation, mass_enrichment={}):

     # this probably needs some naming convention for how the materials will be
     # automatically named for use later
     # OR they get added to a second local/temporary library and we figure out 
     # which library to use for which materials we are mixing
     add_enriched_materials_to_library(material_library, mass_enrichment)

     # the rest of mix_by_volume() as it already eixsts
```